apps/swirl-docs/src/components/CodePreview/CodePreviewRequestString.tsx

```diff
@@ -1,16 +1,18 @@
 import { useCodePreviewContext } from "./CodePreviewContext";
 
 export function APIEndpointHeader() {
-  const { request } = useCodePreviewContext();
+  const { codeExample } = useCodePreviewContext();
```

Comment:
I get your point regarding the usage of props over context, especially given our current component hierarchy.

However, I've chosen to use the Context API because it provides a more flexible way of sharing state across components without prop drilling. This method allowed for decoupling components a bit, which I believe improves readability and maintenance.

Of course, you're absolutely right that as our application evolves, we may find opportunities to refine this approach and revert back to props where it makes sense.

Reply:
It's fine that way. But I wouldn't call this instance prop drilling. It's mostly two component levels here.